### ISSUE TITLE
Aborting a transition clears the resolved model

### ISSUE THREAD

**Issue by btecu** [emberjs/ember.js#14721]
When you abort a transition in `afterModel` and go to an intermediate route, the model for the previous route does not exist, although it was resolved.

https://ember-twiddle.com/db66247a084b6028d31c29ef856d7301 (See the console log)

**Comment by pixelhandler** [comment]
@locks is it possible that the model is not kept on the application route since setup controller was never reached?

**Comment by btecu** [comment]
@pixelhandler No, that's not the case. You can test this easily but writing an empty `setupController()` (don't call `super` to set the `model` on the `controller`). You'll see that `modelFor` still returns.

**Comment by btecu** [comment]
Any ideas on how to work around this?

**Comment by pixelhandler** [comment]
@btecu do you still have this issue, or did you find a work around?

**Comment by btecu** [comment]
@pixelhandler still have the issue. See #16716.

**Comment by pixelhandler** [comment]
Noted that his has a PR for the test, #16716

**Comment by btecu** [comment]
I talked to @rwjblue about this in person and to add a bit more context, in the application model I load some meta data that is only ever going to be loaded once for the lifetime of the application.

The issue appears when a user is loading say `/account` which is an authenticated route. In `beforeModel` (`routes/account.js`) I check whether the session is authenticated or not. If it's not, I abort the transition and put it in the session service to retry it later once the user authenticates, and finally I do a `intermediateTransitionTo('login')` to allow the user authenticate.

Somehow this causes the `application` route model to be **not** resolved.
The model is actually in the store because the workaround (which has it's own issues) is to do something like:
```
// https://github.com/emberjs/ember.js/issues/14721
// let application = this.modelFor('application');
let application = this.store.peekAll('application-model').get('firstObject');
```

**Comment by pixelhandler** [comment]
@btecu glad you found a work around :)
